lib/backends/gcp-secrets-manager-backend.js

```diff
@@ -47,12 +45,32 @@ class GCPSecretsManagerBackend extends KVBackend {
     const version = await this._client.accessSecretVersion({
       name: 'projects/' + projectId + '/secrets/' + key + '/versions/' + secretVersion
     })
-    const secret = { value: version[0].payload.data.toString('utf8') }
+    const secret = version[0].payload.data.toString('utf8')
     // Handle binary files - this is useful when you've stored a base64 encoded string
     if (keyOptions && keyOptions.isBinary) {
-      return Buffer.from(secret.value, 'base64')
+      return Buffer.from(secret, 'base64')
     }
-    return JSON.stringify(secret)
+
+    if (keyOptions && keyOptions.property) {
+      let parsedValue
+
+      try {
+        parsedValue = JSON.parse(secret)
+      } catch (e) {
+        this._logger.warn(`Failed to JSON.parse value for '${key}',` +
+        ' please verify that your secret value is correctly formatted as JSON.' +
+        ` To use plain text secret remove the 'property: ${keyOptions.property}'`)
+        return
+      }
+
+      if (!(keyOptions.property in parsedValue)) {
+        throw new Error(`Could not find property ${keyOptions.property} in ${key}`)
+      }
+
+      return parsedValue[keyOptions.property]
+    }
```

Comment:
Shouldnt be any need to handle this here, and would probably even cause issues to do so

```suggestion
```

This is handled here:
https://github.com/godaddy/kubernetes-external-secrets/blob/945d1fdb44a17e98fb0281d77fa65c8e0048456d/lib/backends/kv-backend.js#L30-L50

Reply:
Ok, I got it ...

My bad, didn't see the extends KVBackend